### ISSUE TITLE
Unwanted /data volume hogs system

### ISSUE THREAD

**Issue by janaxhell** [rogerfar/rdt-client#213]
Yesterday :latest build broke something and at the end of the day it was solved with 2.0.21. But today I see a new problem.

I have these volumes set:

```
/srv/dev-disk-by-uuid-5b67514d-485e-4306-873e-b1cbb54ccf99/Config/rdtclient:/config
/srv/dev-disk-by-uuid-2628F1C228F19153/data/db:/data/db
/srv/dev-disk-by-uuid-2628F1C228F19153/data/rdtclient:/data/rdtclient
/srv/dev-disk-by-uuid-A870CA6B70CA3FB4/data/rdtclient:/data2/rdtclient
```
But each time I recreate RDTclient, it creates this one:

`/srv/dev-disk-by-uuid-2628F1C228F19153/data/rdtclient:/data`

that is redundant because I have already set
`/srv/dev-disk-by-uuid-2628F1C228F19153/data/rdtclient:/data/rdtclient`
and I specifically don't want it because it makes rdtclient check permissions of thousands of files inside /data.
But no matter what, if I remove it from container, it reappears at start.
I created 
```
[...]:/data/rdtclient
[...]:/data2/rdtclient
```
that are 2 separate drives to limit permission check to just /rdtclient folder. And it worked fine until 2.0.19.
Now it hogs the system again, but this time log says nothing about it (like "Setting Permissions"), I can only hear the HDD thrashing and see rdclient abusing CPU, RAM and SystemLoad from Grafana for one movie waiting to be imported in Radarr for no less than 20 minutes.  

My compose looks like this

```
version: '3.3'
services:
    rdtclient:
        container_name: rdtclient
        environment:
        - PUID=998
        - PGID=100
        volumes:
            - '/srv/dev-disk-by-uuid-2628F1C228F19153/data/rdtclient:/data/rdtclient'
            - '/srv/dev-disk-by-uuid-A870CA6B70CA3FB4/data/rdtclient:/data2/rdtclient'
            - '/srv/dev-disk-by-uuid-2628F1C228F19153/data/db:/data/db'
            - '/srv/dev-disk-by-uuid-5b67514d-485e-4306-873e-b1cbb54ccf99/Config/rdtclient:/config'
        image: rogerfar/rdtclient
        restart: always
        logging:
            driver: json-file
            options:
                max-size: 10m
        ports:
            - '6500:6500'
```
As you can see I haven't set /srv/dev-disk-by-uuid-2628F1C228F19153/data/rdtclient:/data anywhere, nevertheless it reappears in container volumes as shown in picture below.

**What version are you using?**
2.0.21
**Wat OS are you running?**
OMV6
**Are you using Docker or as a service?**
Docker
**Please attach a log file here with the log setting set to debug**

![vivaldi_mlDfKCLKXQ](https://user-images.githubusercontent.com/55061479/224287998-b6b1162c-5742-4a59-9aa3-5c80c18bcded.png)

EDIT Happened again with Sonarr. This definitely didn't happen before 2.0.21

EDIT 2 I reverted back to 2.0.19, but it keeps happening. This is the log

```
[custom-init] No custom services found, skipping...
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service 00-legacy: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service 00-legacy successfully started
cont-init: info: running /etc/cont-init.d/01-envfile
cont-init: info: /etc/cont-init.d/01-envfile exited 0
cont-init: info: running /etc/cont-init.d/01-migrations
[migrations] started
[migrations] no migrations found
cont-init: info: /etc/cont-init.d/01-migrations exited 0
cont-init: info: running /etc/cont-init.d/10-adduser
-------------------------------------
          _         ()
         | |  ___   _    __
         | | / __| | |  /  \
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/
Brought to you by linuxserver.io
-------------------------------------
To support LSIO projects visit:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------
User uid:    998
User gid:    100
-------------------------------------
cont-init: info: /etc/cont-init.d/10-adduser exited 0
cont-init: info: running /etc/cont-init.d/30-config
Setting permissions
chown: WARNING: Circular directory structure.
This almost certainly means that you have a corrupted file system.
NOTIFY YOUR SYSTEM MANAGER.
The following directory is part of the cycle:
  /data/rdtclient
cont-init: info: /etc/cont-init.d/30-config exited 1
cont-init: info: running /etc/cont-init.d/99-custom-files
[custom-init] No custom files found, skipping...
cont-init: info: /etc/cont-init.d/99-custom-files exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service init-mods: starting
s6-rc: info: service init-mods successfully started
s6-rc: info: service init-mods-package-install: starting
s6-rc: info: service init-mods-package-install successfully started
s6-rc: info: service init-mods-end: starting
s6-rc: info: service init-mods-end successfully started
s6-rc: info: service init-services: starting
s6-rc: info: service init-services successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun rdtclient (no readiness notification)
s6-rc: info: service legacy-services successfully started
s6-rc: info: service 99-ci-service-check: starting
[ls.io-init] done.
s6-rc: info: service 99-ci-service-check successfully started
Changing to /app folder
Starting rtdclient
[13:03:29 WRN] Using an in-memory repository. Keys will not be persisted to storage.
[13:03:30 WRN] Neither user profile nor HKLM registry available. Using an ephemeral key repository. Protected data will be unavailable when application exits.
[13:03:30 WRN] No XML encryptor configured. Key {40eb9666-a151-428a-9eef-ad35863f181b} may be persisted to storage in unencrypted form.
[13:03:30 WRN] Starting host on version 2.0.19.0
[13:04:07 WRN] The query uses the 'First'/'FirstOrDefault' operator without 'OrderBy' and filter operators. This may lead to unpredictable results.
[13:04:07 WRN] 401: /Api/Authentication/IsLoggedIn
```

**Comment by janaxhell** [comment]
Yes, that's the folder that I want to use, but rdtclient duplicates as /data. 
I want to use /rdtclient folders in /data and /data2 so that there is no file to check permission, because they're empty as soon as *arrs import the content, but rdtclient keeps creating a third folder that links /data/rdtclient to /data.

**Comment by rogerfar** [comment]
And in RDT, what paths do you have configured under the `Download Client` settings?

**Comment by janaxhell** [comment]
Download path: /data2/rdtclient
Mapped path: /data2/rdtclient
I haven't set /data/rdtclient because I have set "music" as category for RDT in Lidarr, so when it finds an album, it looks for the remote folder /data2/music.

I admit that the impossibility to set multiple folders from GUI confuses me a lot. Nevertheless until 2.0.19 this setup worked fine.

**Comment by janaxhell** [comment]
So, since RDTclient keeps creating .../data/rdtclient:/data no matter what, is it possible to map 2 separate HDDs so that when Lidarr triggers, rdtclient downloads to HDD1 and when Radarr and Sonarr trigger, rdtclient downloads to HDD2, giving the chance to all *arrs to make atomic move in place without sending files across drives? 
In settings I can only set one Mapped path and one Download path, but I need at least 2 Download paths.
I have tried to map them as in compose above, but I can't set the same in GUI settings.

**Comment by rogerfar** [comment]
Could be a result of the permission issue discussed in #214. Try the latest image and make sure your permissions are set correctly.

**Comment by janaxhell** [comment]
No, all I get with or without PUID:PGID 1000:1000 is 
`s6-rc-compile: fatal: cyclic service dependency reached from legacy-services and involving legacy-services`
I can only deploy 2.0.19 but I MUST delete 
`/srv/dev-disk-by-uuid-2628F1C228F19153/data/rdtclient:/data`
otherwise container refuses to be deployed AND after it is deployed, that folder is created again by RDTclient, not by me.
Basically it says there is a loop, but it is it that creates the loop.

**Comment by rogerfar** [comment]
So 2.0.25 is still not working correctly then?

**Comment by janaxhell** [comment]
I have just tried to deploy it with 1000:1000 and got the above error. 
I will try again with explicit 2.0.25, I used :latest

**Comment by janaxhell** [comment]
Ok, with 2.0.25 it deploys with no modification to PUID:PGID, but it still says
```
chown: WARNING: Circular directory structure.
This almost certainly means that you have a corrupted file system.
NOTIFY YOUR SYSTEM MANAGER.
The following directory is part of the cycle:
  /data/rdtclient
chown: WARNING: Circular directory structure.
This almost certainly means that you have a corrupted file system.
NOTIFY YOUR SYSTEM MANAGER.
The following directory is part of the cycle:
  /data/rdtclient
Changing to /app folder
Starting rtdclient
s6-rc: warning: unable to start service init-rdt-client: command exited 1
[15:54:19 WRN] Using an in-memory repository. Keys will not be persisted to storage.
[15:54:19 WRN] Neither user profile nor HKLM registry available. Using an ephemeral key repository. Protected data will be unavailable when application exits.
[15:54:20 WRN] No XML encryptor configured. Key {0fe35c48-2c89-4475-9f9e-49b8625b0269} may be persisted to storage in unencrypted form.
[15:54:20 WRN] Starting host on version 2.0.25.0
[15:54:26 WRN] The query uses the 'First'/'FirstOrDefault' operator without 'OrderBy' and filter operators. This may lead to unpredictable results.
[15:54:26 WRN] 401: /api/v2/torrents/info
[15:54:39 WRN] 401: /api/v2/torrents/info
[15:54:40 WRN] 401: /api/v2/torrents/info
[15:55:13 WRN] 401: /Api/Authentication/IsLoggedIn

```
for the folder 
`/srv/dev-disk-by-uuid-2628F1C228F19153/data/rdtclient:/data`
that it creates on its own every time.

**Comment by rogerfar** [comment]
Is this still an issue?

**Comment by janaxhell** [comment]
Well, that's weird. It worked fine, I forgot about this issue until you replied. So I checked by restarting the container: no issue.
But when I checked the GUI, I saw the notification about me using v.2.0.37 while v2.0.38 was available.
So I recreated the container to :latest (=2.0.38) and it broke, it gave this message (tried also to restart):

```
s6-rc-compile: fatal: unable to read /etc/s6-overlay/s6-rc.d/init-addusser/type: No such file or directory
s6-rc: fatal: unable to take locks: No such file or directory
s6-linux-init-shutdownd: warning: /run/s6/basedir/scripts/rc.shutdown exited 111
s6-rc-compile: fatal: unable to read /etc/s6-overlay/s6-rc.d/init-addusser/type: No such file or directory
```
So I went back to to 2.0.37 and everything went back to normal and working, with this message:

```
Starting rtdclient
s6-rc: warning: unable to start service init-rdt-client: command exited 1
[09:46:46 WRN] Using an in-memory repository. Keys will not be persisted to storage.
[09:46:46 WRN] Neither user profile nor HKLM registry available. Using an ephemeral key repository. Protected data will be unavailable when application exits.
[09:46:46 WRN] No XML encryptor configured. Key {d09646c8-1cfd-4865-892a-60d7fc383c4b} may be persisted to storage in unencrypted form.
[09:46:46 WRN] Starting host on version 2.0.37.0
[09:47:14 WRN] 403: /Api/Authentication/IsLoggedIn
[09:47:19 WRN] 101: /hub
```
So until 2.0.37, no issue, but 2.0.38 is totally non-working.

Ironically yesterday Uptobox was seized, so RD & AD are not working, making rdtclient useless at the moment.

**Comment by vromero** [comment]
I suspect it might be related to https://github.com/rogerfar/rdt-client/blob/master/Dockerfile#L106

**Comment by rogerfar** [comment]
@janaxhell is this still an issue with the latest versions?

**Comment by janaxhell** [comment]
> @janaxhell is this still an issue with the latest versions?

So far I had no problem using it since Alldebrid fixed the Uptobox problem. But since you asked, I checked the app and noticed it was on 2.0.49 with 2.0.50 available. So I updated to :latest, but when I opened it, it said **Version v2.0.52 of RealDebrid Client was found. You are currently on version v2.0.50.**
So I updated container again, but it's still on 2.0.50 (Portainer notified the successful replacement of the image).
This is weird for 2 reasons: Watchtower did not update it (it does for every other container) and I cannot get past v2.0.50.
My compose hasn't changed, it's the same as above.
